repo_policy_compliance/__init__.py

```diff
@@ -59,6 +60,60 @@ class Report(NamedTuple):
     reason: str | None
 
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def log_check(func: Callable[P, R]) -> Callable[P, R]:
+    """Log before check and result of check.
+
+    Args:
+        func: The function that executes a check.
+
+    Returns:
+        The function where the check is logged before it starts and the results are logged.
+    """
+    check_name = func.__name__
+
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
+        """Replace function.
+
+        Args:
+            args: The positional arguments passed to the method
+            kwargs: The keywords arguments passed to the method
+
+        Returns:
+            The return value after calling the wrapped function.
+        """
+        logging.info("start check '%s'", check_name)
+        result = func(*args, **kwargs)
+        logging.info("check '%s' finished, result: %s", check_name, result)
+        return result
+
+    return wrapper
+
+
+def _setup_logging() -> None:
+    """Initialise logging for check execution."""
+    handler = logging.FileHandler(filename="/dev/stdout")
```

Comment:
If we only want to log to stderr, would the `StreamHandler` and `sys.stderr` be a better choice?

Reply:
Shouldn't errors be logged there? This is supposed to log debug info which sounds like it should go to stdout

Reply:
Yes, I mean `StreamHandler` and `sys.stdout`.

Reply:
I'm not following? Are you suggesting adding another `error` handler that writes to `stderr`?

Reply:
I thinks that `sys.stdout` could be a better alternative to the hard-coded `"/dev/stdout"` string. And the `StreamHandler` could be better alternative for the `FileHandler` since we are output to the stream. It's just a minor suggestion.

Reply:
ah now I'm following

Reply:
Should be fixed now